fiatconnect-api.md

```diff
@@ -391,105 +407,204 @@ The FiatConnect API specifies *two* types of authentication; the first authentic
+cookie is not mandated by this specification, but the server MUST use a consistent name for the session cookie, and SHOULD make a best-effort at choosing a unique name, that
+will not conflict with the session cookies served by other FiatConnect-compliant APIs. It is the client's responsibility to manage session cookies between different FiatConnect
+APIs.
+
+If a client sends a session cookie when calling the `POST /auth/login` endpoint (either to refresh their current, valid session, or to generate a new session if their current one is
+expired), the server MUST ignore the provided cookie. If the authentication request is valid, the server MUST set a new value on the cookie referencing the client's new, valid session,
+and return it to the client.
+
+#### 3.2.1.5. `POST /auth/login`
+
+The `POST /auth/login` endpoint is responsible for verifying signed messages sent by clients, creating and authenticating a user's session, and returning session cookies.
+Note that both the Sign-In With Ethereum standard, and FiatConnect, support authorization for externally owned accounts (EOAs) and contract-owned accounts; The `POST /auth/login`
+endpoint MUST honor login requests for both, as described below.
+
+##### 3.2.1.5.1. Parameters
+
+###### 3.2.1.5.1.1. Request Body
+
+The *plaintext* request body for this endpoint MUST correspond to a signed SIWE message (as described above) in the [EIP-191](https://eips.ethereum.org/EIPS/eip-191)
+Signed Data Standard format, as required by the SIWE specification.
+
+##### 3.2.1.5.2. Responses
+
+###### 3.2.1.5.2.1. HTTP `200`
+
+On success, the server MUST return an HTTP `200`, update the state of the user's session to be authenticated, and return a session cookie.
+
+###### 3.2.1.5.2.2. HTTP `401`
+
+On failure, the server MUST return an HTTP `400`, with the following response body.
+
+```
 {
-	"alg": "ES256",
-	"typ": "JWT"
+	error: `ErrorEnum`,
 }
 ```
 
-Servers MAY support other algorithms, but this is not required.
+##### 3.2.1.5.3. Semantics
+
+This endpoint performs validation on a signed SIWE-compliant message. In addition to checking that the signature itself is valid and corresponds to the included
+address, it also performs various safety and security checks on the contents of the message. If all checks pass, the server updates the client's session to be authenticated,
+and returns a `200`.
 
-#### 3.2.1.2. JWT Payload
+###### 3.2.1.5.3.1. Success
 
-Servers MUST recognize the following *registered claims* within the JWT payload.
+On success, the server MUST respond with a `200` status code. The server MUST only respond successfully after performing the following checks:
 
-* `exp`, or *expiration time*
-* `sub`, or *subject*
-* `iss`, or *issuer*
+* A server MUST verify that the signature of the request body is valid and corresponds to the `address` field included in the signed message. If the signature does not
```

Comment:
We should make it more clear that the they should start by making sure the whole message is valid according to the ABNF specified in SIWE.

Reply:
Good callout; specified this!